src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +430,1174 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
             }
         }
 
-        private static string[] GetNullableDataKeysAsStrings(ConcurrentDictionary<object, int> nullableData) =>
-            nullableData.Keys.Select(key => GetNullableDataKeyAsString(key)).OrderBy(key => key).ToArray();
+        private readonly struct NullableDirectives
+        {
+            internal readonly string[] Directives;
+            internal readonly NullableContextState.State ExpectedWarningsState;
+            internal readonly NullableContextState.State ExpectedAnnotationsState;
+
+            internal NullableDirectives(string[] directives, NullableContextState.State expectedWarningsState, NullableContextState.State expectedAnnotationsState)
+            {
+                Directives = directives;
+                ExpectedWarningsState = expectedWarningsState;
+                ExpectedAnnotationsState = expectedAnnotationsState;
+            }
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_01()
+        {
+            var projectSettings = new[]
+            {
+                (NullableContextOptions?)null,
+                NullableContextOptions.Disable,
+                NullableContextOptions.Warnings,
+                NullableContextOptions.Annotations,
+                NullableContextOptions.Enable,
+            };
+            var nullableDirectives = new[]
+            {
+                new NullableDirectives(new string[0], NullableContextState.State.Unknown, NullableContextState.State.Unknown),
```

Comment:
>NullableContextState.State.Unknown [](start = 54, length = 34)

Consider using named parameters for these arguments, would make it much easier to comprehend what the tests are doing.

Reply:
I've added a comment above the elements. Named parameters will make these lines longer and I suspect more difficult to read.

---
In reply to: [546881283](https://github.com/dotnet/roslyn/pull/49953#discussion_r546881283) [](ancestors = 546881283)